plugins/users/settings/delete.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const schema = require('screwdriver-data-schema');
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/users/settings',
+    options: {
+        description: 'Reset user settings',
+        notes: 'Reset user settings',
+        tags: ['api', 'users'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+
+        handler: async (request, h) => {
+            const { userFactory } = request.server.app;
+            const { scmContext, username } = request.auth.credentials;
+            const { settings } = request.payload;
+            const user = await userFactory.get({ username, scmContext });
+
+            if (!user) {
+                throw boom.notFound('User does not exist');
+            }
+
+            return user.updateSettings(settings).then(results => {
```

Comment:
Just checked `user.removeSettings` does not exist, please add one in https://github.com/screwdriver-cd/models/blob/90de0b119ff542dba128ea690a1530896515d940/lib/user.js#L120-L126, and call it here.

It will be easier to modify `user.updateSettings` later if `user.removeSettings` is used here.